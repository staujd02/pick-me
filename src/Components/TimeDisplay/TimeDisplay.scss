@import '../../Assets/styling/mixins.scss';
@import '../../Assets/styling/palette.scss';

.time-display.MuiButton-root{
    @include alpha-background-color(rgba($color1, $airy), $color2);

    position: absolute;
    right: 15px;
    top: 15px;
    height: 70px;
    width: auto;
    padding-left: 20px;
    padding-right: 20px;
    border-radius: 40px;
    cursor: pointer;
}

.time-display.MuiButton-root:hover{
    @include alpha-background-color(rgba($color1, $light), $color2);
}

.time-box{
    @include alpha-attribute('color', rgba($color4, $heavy), $color5);
    font-family: 'Trebuchet MS', 'Lucida Sans Unicode', 'Lucida Grande', 'Lucida Sans', Arial, sans-serif;
    font-size: 24px;
    text-align: center;
}