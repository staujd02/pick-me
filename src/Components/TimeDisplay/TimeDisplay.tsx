import React from 'react';
import './TimeDisplay.scss';
import { Button } from '@material-ui/core';

class TimeDisplay extends React.Component<TimeDisplayProps>{

    render(){
        const { time, onClick } = this.props;
        const secondsRemaining = this.calcSeconds(time);
        const minutesRemaining = this.calcMinutes(time);
        return (
            <Button onClick={onClick} color={"inherit"} className="time-display">
                <div className="time-box">{`${minutesRemaining}:${secondsRemaining}`}</div>
            </Button>
        );
    }

    private calcMinutes(milSecondsRemaining: number) {
        return this.format(Math.floor(milSecondsRemaining / 60000));
    }

    private calcSeconds(milSecondsRemaining: number) {
        return this.format(Math.floor((milSecondsRemaining % 60000) / 1000));
    }

    private format(numberToFormat: number): string {
        var str = "" + numberToFormat;
        var pad = "00";
        return pad.substring(0, pad.length - str.length) + str;
    }
}

export default TimeDisplay;