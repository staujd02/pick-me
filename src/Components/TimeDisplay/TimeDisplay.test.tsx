import React from 'react';
import TimeDisplay from './TimeDisplay';
import { ShallowWrapper, shallow } from 'enzyme';

describe("The time display", () => {

    let wrapper: ShallowWrapper;
    let onClick = jest.fn();

    beforeEach(() => {
        wrapper = shallow(<TimeDisplay onClick={onClick} time={getTime()}/>);
    });

    it('renders correctly', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    describe("when the user clicks the time display", () => {

        beforeEach(() => {
            wrapper.find('.time-display').simulate('click');
        });

        it("fires the interaction handler", () => {
            expect(onClick).toHaveBeenCalled();
        });
    });

    function getTime() {
        let sixtyMinutesInMilli = 60 * 60 * 1000;
        let thirtySecondsInMilli = 3 * 1000;
        return sixtyMinutesInMilli + thirtySecondsInMilli;
    }

});