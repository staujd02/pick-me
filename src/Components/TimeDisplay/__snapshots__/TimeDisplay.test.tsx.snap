// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The time display renders correctly 1`] = `"<button class=\\"MuiButtonBase-root MuiButton-root time-display MuiButton-text MuiButton-colorInherit\\" tabindex=\\"0\\" type=\\"button\\"><span class=\\"MuiButton-label\\"><div class=\\"time-box\\">60:03</div></span></button>"`;
